presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -690,7 +694,13 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
 
         checkTableIsWritable(table.get());
 
-        HiveStorageFormat hiveStorageFormat = extractHiveStorageFormat(table.get());
+        HiveStorageFormat hiveStorageFormat;
+        if (respectTableFormat) {
```

Comment:
No, this option was only intended to handle the case where Presto is configured to write a certain format and you're doing an insert on a table that declares a different format. It was never meant to limit the formats users are allowed to write.

For the case you're describing, if that's a problem, then the engine should protect against that (by disallowing writing ORC tables, reducing parallelism, etc)